rule/var_naming.go

```diff
@@ -279,7 +285,7 @@ func getList(arg any, argName string) ([]string, error) {
 	for _, v := range args {
 		val, ok := v.(string)
 		if !ok {
-			return nil, fmt.Errorf("invalid %s values of the var-naming rule. Expecting slice of strings but got element of type %T", val, arg)
+			return nil, fmt.Errorf("invalid %v values of the var-naming rule. Expecting slice of strings but got element of type %T", v, arg)
```

Comment:
Here another oddity, that can be seen in the unit test file you added

```
wantErr:   errors.New("invalid argument to the var-naming rule. Expecting a allowlist of type slice with initialisms, got int"), 
(...)
wantErr:   errors.New("invalid 123 values of the var-naming rule. Expecting slice of strings but got element of type []interface {}"),
```

The first error message mention the name of the argument, the second doesn't

Maybe something like this, that remove the "a", because "a allowList" sounds bad.

```suggestion
			return nil, fmt.Errorf("invalid %v values of the var-naming rule. Expecting '%v' parameter to be a slice of strings but got element of type %T", argName, v, arg)
```